Protocol.md

```diff
@@ -279,6 +279,50 @@ The recipient:
 
   These script pub key forms include only standard forms accepted by the wider set of deployed Bitcoin clients in the network, which increase the chances of successful propagation to miners.
 
+## Second Layer Interaction
+
+Second Layer Interaction consist of second layer "features" which parties can opt-in to using. These messages are not needed to facilitate construction of a DLCs.
+
+### The `close_dlc` Message
+
+This message contains information about mutual close, which allows the counterparty
+to broadcast a mutual closing transaction.
+
+1. type: ? (`close_dlc_v0`)
+2. data:
+   * [`32*byte`:`contract_id`]
+   * [`signature`:`close_signature`]
+   * [`u64`:`offerPayoutSatoshis`]
+   * [`u64`:`acceptPayoutSatoshis`]
+   * [`u64`:`fund_input_serial_id`]
+   * [`u16`:`num_funding_inputs`]
+   * [`num_funding_inputs*funding_input`:`funding_inputs`]
+
+
+`payout_spk` and `payout_serial_id` from `offer_dlc` as well as `payout_spk` and `payout_serial_id` from `accept_dlc` should be used for constructing the close transaction
+
+`fund_input_serial_id` is a randomly chosen number which uniquely identifies the funding input of the initiating party.
+Inputs in the closing transaction will be sorted by `fund_input_serial_id` and `input_serial_id` in `funding_inputs`.
+
+`funding_inputs` are extra inputs to mutual close to avoid free option, and in the future revocation mechanism will be introduced
+
+#### Requirements
+
+The sender MUST:
+
+  - set `contract_id` by exclusive-OR of the `funding_txid`, the `funding_output_index` and the `temporary_contract_id` from the `offer_dlc` and `accept_dlc` messages.
+  - set `close_signature` to the valid signature, using its `funding_pubkey` for the close transaction, as defined in the [transaction specification](Transactions.md#close-transaction).
+
+The recipient:
+
+  - if any input in `funding_inputs` is not a BIP141 (Segregated Witness) input.
```

Comment:
I think it's better anyway but I'm not sure it's a strong requirement in this case.

Reply:
Good point, because malleability attacks aren't a concern since there's no transaction to spend the outputs of dlc close. I'll change to `MAY ignore the message`

Reply:
Isn't this implied that all inputs are BIP141 compliant as the we require dlc funding outputs to be BIP141?

Reply:
That's about the inputs added to the closing transactions, not the funding outputs, so unless I'm mistaken I don't think we need specific constraints on these (even though it's probably a good idea).

Reply:
So your closing transaction can have unrelated inputs to the in progress DLC that you are trying to close?

Reply:
Yes in order to be able to double spend them to make the closing transaction invalid if your counter party doesn't broadcast it in a timely manner.

Reply:
That's correct. The purpose of these unrelated inputs is to enable the initiator of the mutual close to have the ability to nullify the close transaction by spending those inputs in the event that counterparty doesn't immediately act on the mutual close.